torchao/prototype/low_bit_optim/subclass_fp8.py

```diff
@@ -62,15 +77,22 @@ def dequantize(self, output_dtype=None):
         float_data = self.codes.float()
         float_data = float_data.view(-1, self.block_size) * self.scale.view(-1, 1)
 
+        if self.k:
```

Comment:
I think this will error out when `k` is a tensor. `if self.k is not None` should be better.

Reply:
understood

Reply:
is it because this is ambiguous ?